control-plane-operator/hostedclusterconfigoperator/operator/config.go

```diff
@@ -87,8 +88,17 @@ func Mgr(ctx context.Context, cfg, cpConfig *rest.Config, namespace string, hcpN
 		panic(fmt.Sprintf("failed to create client: %v", err))
 	}
 	hcp := resourcemanifests.HostedControlPlane(namespace, hcpName)
-	if err = ct.Get(ctx, client.ObjectKeyFromObject(hcp), hcp); err != nil {
-		panic(fmt.Sprintf("unable to get HCP: %v", err))
+
+	// Retry getting HCP to handle transient connectivity issues
+	err = wait.PollUntilContextTimeout(ctx, 2*time.Second, 30*time.Second, true, func(ctx context.Context) (bool, error) {
+		if err := ct.Get(ctx, client.ObjectKeyFromObject(hcp), hcp); err != nil {
+			fmt.Printf("Failed to get HCP %s/%s, retrying: %v\n", namespace, hcpName, err)
```

Comment:
Sorry I missed this before, but shouldn't we use a  Logger instead of `Printf` ?

Reply:
We don’t have logger configured there yet, it’s bootstrapping the manager which launches the controllers. This is also why we panic instead of handle the error in other way.